tests/core/test_callable_utils.py

```diff
@@ -0,0 +1,1155 @@
+
+    sql_body = generate_sql_function_body(try_except_func, "test_catalog", "test_schema")
+    
+    expected_sql = """
+CREATE OR REPLACE FUNCTION test_catalog.test_schema.try_except_func(a INTEGER COMMENT 'First number', b INTEGER COMMENT 'Second number')
+RETURNS INTEGER
+LANGUAGE PYTHON
+COMMENT 'A function with try-except block.'
+AS $$
+    try:
+        return a + b
+    except Exception as e:
+        raise ValueError(f"Invalid operation") from e
+$$;
+    """
+    assert sql_body.strip() == expected_sql.strip()
+
+def test_function_with_multiple_return_paths():
+    def multiple_return_func(a: int) -> str:
+        """A function with multiple return paths."""
+        if a > 0:
+            return "Positive"
+        else:
+            return "Negative"
+
+    sql_body = generate_sql_function_body(multiple_return_func, "test_catalog", "test_schema")
+    
+    expected_sql = """
+CREATE OR REPLACE FUNCTION test_catalog.test_schema.multiple_return_func(a INTEGER COMMENT 'Parameter a')
+RETURNS STRING
+LANGUAGE PYTHON
+COMMENT 'A function with multiple return paths.'
+AS $$
+    if a > 0:
+        return "Positive"
+    else:
+        return "Negative"
+$$;
+    """
+    assert sql_body.strip() == expected_sql.strip()
+
+def test_function_returning_none():
+    def func_returning_none(a: int) -> None:
+        """
+        A function that returns None.
+
+        Args:
+            a: An integer
+        """
+        return None
+
+    with pytest.raises(ValueError, match=" in return type for function 'func_returning_none': <class 'NoneType'>. Unsupported return type: <class 'NoneType'>."):
+        generate_sql_function_body(func_returning_none, "test_catalog", "test_schema")
+
+def test_function_returning_any():
+    def func_returning_any(a: int) -> Any:
+        """
+        A function that returns Any type.
+
+        Args:
+            a: An integer
+        """
+        return a
+
+    with pytest.raises(ValueError, match="Error in return type for function 'func_returning_any': typing.Any. 'Any' type is not supported. Please specify a concrete return type."):
+        generate_sql_function_body(func_returning_any, "test_catalog", "test_schema")
+
+def test_function_returning_union():
+    def func_returning_union(a: int) -> Union[int, str]:
+        """
+        A function that returns a Union of int and str.
+
+        Args:
+            a: An integer
+        
+        Returns:
+            Union[int, str]: Either an integer or a string
+        """
+        if a > 0:
+            return a
+        return str(a)
+
+    with pytest.raises(ValueError, match=re.escape("Error in return type for function 'func_returning_union': typing.Union[int, str]. Union types are not supported in return types.")):
+        generate_sql_function_body(func_returning_union, "test_catalog", "test_schema")
+
+# ---------------------------
+# Tests for Input Types
+# ---------------------------
+
+def test_function_with_list_input():
```

Comment:
As we're only testing different input types, we can parametrize them? Not a blocker, we can refactor in a follow-up PR, this PR is too much :)

Reply:
Yeah, I'm voting for a followup for optimizing tests